e2e/test/scenarios/dashboard-cards/click-behavior.cy.spec.js

```diff
@@ -288,6 +298,93 @@ describe("scenarios > dashboard > dashboard cards > click behavior", () => {
       });
     });
 
+    it("allows setting dashboard tab with parameter as custom destination", () => {
+      cy.createDashboard(
+        {
+          ...TARGET_DASHBOARD,
+          parameters: [DASHBOARD_FILTER_TEXT],
+        },
+        {
+          wrapId: true,
+          idAlias: "targetDashboardId",
+        },
+      );
+      cy.get("@targetDashboardId").then(targetDashboardId => {
+        cy.request("PUT", `/api/dashboard/${targetDashboardId}/cards`, {
+          cards: [],
+          tabs: [FIRST_TAB, SECOND_TAB, THIRD_TAB],
+        });
+      });
+      cy.createQuestionAndDashboard({ questionDetails }).then(
+        ({ body: card }) => {
+          visitDashboard(card.dashboard_id);
+          cy.wrap(card.dashboard_id).as("dashboardId");
+        },
+      );
+
+      editDashboard();
+
+      getDashboardCard().realHover().icon("click").click();
+      addDashboardDestination();
+      cy.get("aside")
+        .findByLabelText("Select a dashboard tab")
+        .should("have.value", FIRST_TAB.name)
+        .click();
+      cy.findByRole("listbox").findByText(SECOND_TAB.name).click();
+      cy.get("aside").findByText("No available targets").should("not.exist");
+      addTextParameter();
+      cy.get("aside").button("Done").click();
+
+      saveDashboard();
+
+      clickLineChartPoint();
+      cy.findAllByTestId("field-set")
+        .should("have.length", 1)
+        .should("contain.text", POINT_COUNT);
+      cy.get("@targetDashboardId").then(targetDashboardId => {
+        cy.location().should(({ pathname, search }) => {
+          expect(pathname).to.equal(`/dashboard/${targetDashboardId}`);
+          expect(search).to.equal(
+            `?tab=${SECOND_TAB_SLUG}&${DASHBOARD_FILTER_TEXT.slug}=${POINT_COUNT}`,
+          );
+        });
+      });
+
+      cy.log("click behavior should still work after removing target tab");
+      editDashboard();
+      deleteTab(SECOND_TAB.name);
+      saveDashboard();
+
+      cy.get("@dashboardId").then(dashboardId => {
+        visitDashboard(dashboardId);
+      });
+
+      editDashboard();
+      getDashboardCard().realHover().icon("click").click();
+      cy.get("aside")
+        .findByLabelText("Select a dashboard tab")
+        .should("have.value", FIRST_TAB.name)
+        .click();
```

Comment:
I don't see the point of clicking on the tab select here, if you're not going to change it. And you already assert that it has the value of the first tab name.
```suggestion
```

![image](https://github.com/metabase/metabase/assets/1937582/d7081924-5dbb-464f-bd4d-45ddc1fac55a)

Reply:
Oh good catch, I wanted to add an assertion here that `SECOND_TAB` is no longer in the dropdown: 43a3826a7650da13c83338fdd9a65412ba8b705